src/components/dictionaryConcepts/containers/CreateConcept.jsx

```diff
@@ -253,10 +258,30 @@ export class CreateConcept extends Component {
   updateEventListener = (event) => {
     const { tabIndex, name, value } = event.target;
     const { mappings } = this.state;
-    mappings[tabIndex][name] = value;
+    // if the ciel source URL was provided, use just the name
+    if (value.toUpperCase().match(INTERNAL_MAPPING_DEFAULT_SOURCE)) {
```

Comment:
Why just use the name in this case?

Reply:
@dkayiwa When the user types a source and selects from the dropdown, what the app takes is the URL to that source and not the displayed name (because the backend takes a URL).
So what I'm doing here is checking to see if that value (URL) contains CIEL (a special case) so that I can save the CIEL as it is and not its URL.

Reply:
Contains CIEL or is equal to CIEL?

Reply:
If the URL of the source `contains` CIEL. All other sources will not contain CIEL

Reply:
So what about if the source contains something like CIELBRUCE?

Reply:
It will take the value CIEL. Let me filter by the entire URL to prevent that from happening.

Reply:
@dkayiwa I have made the correction

Reply:
Have you tested the application again with this change to confirm that it still works?

Reply:
Yes. I manually tested both `Create Concept` and `Edit Concept` with the concept name `HS1` in the `Kampala Dictionary`